python-backend/app/api/permits/permit_amendment/resources/permit_amendment.py

```diff
@@ -21,7 +44,89 @@ def get(self, permit_guid=None, permit_amendment_id=None):
             if permit:
                 permit_amendments = PermitAmendment.find_by_permit_id(permit.permit_id)
                 if permit_amendments:
-                    result = {'amendments': [x.json() for x in permit_amendments]}
-                    return result
+                    return [x.json() for x in permit_amendments]
 
         return self.create_error_payload(404, 'Permit amendment(s) not found'), 404
+
+    @api.doc(params={
+        'permit_amendment_guid': 'Permit amendment guid.',
+        'permit_guid': 'Permit GUID'
+    })
+    @requires_role_mine_create
+    def post(self, permit_guid=None, permit_amendment_guid=None):
+        if not permit_guid:
+            return self.create_error_payload(400, 'Permit_guid must be provided'), 400
+        if permit_amendment_guid:
+            return self.create_error_payload(400, 'unexpected permit_amendement_id'), 400
+
+        permit = Permit.find_by_permit_guid(permit_guid)
+        if not permit:
+            return self.create_error_payload(404, 'permit does not exist'), 404
+
+        data = self.parser.parse_args()
+
+        received_date = data.get('received_date')
+        issue_date = data.get('issue_date')
+        authorization_end_date = data.get('authorization_end_date')
+
+        new_pa = PermitAmendment.create(
+            permit,
+            received_date,
+            issue_date,
+            authorization_end_date,
+            self.get_create_update_dict(),
+            save=True)
+
+        return new_pa.json()
+
+    @api.doc(params={
+        'permit_amendment_guid': 'Permit amendment guid.',
+        'permit_guid': 'Permit GUID'
+    })
+    @requires_role_mine_create
+    def put(self, permit_guid=None, permit_amendment_guid=None):
+        if not permit_amendment_guid:
+            return self.create_error_payload(400, 'permit_amendment_id must be provided'), 400
+        pa = PermitAmendment.find_by_permit_amendment_guid(permit_amendment_guid)
+        if not pa:
+            return self.create_error_payload(404, 'permit amendment not found'), 404
+
+        data = self.parser.parse_args()
+        current_app.logger.info(f'updating {pa} with >> {data.keys()}')
+
+        if 'received_date' in data.keys():
+            pa.received_date = data.get('received_date')
+        if 'issue_date' in data.keys():
+            pa.issue_date = data.get('issue_date')
+        if 'authorization_end_date' in data.keys():
+            pa.authorization_end_date = data.get('authorization_end_date')
+        if 'permit_amendment_status_code' in data.keys():
+            pa.permit_amendment_status_code = data.get('permit_amendment_status_code')
+        if 'permit_amendment_type_code' in data.keys():
+            pa.permit_amendment_type_code = data.get('permit_amendment_type_code')
```

Comment:
Could you test for throwing an `AssertionError` here? I have a feeling that the `validates` methods are executed when values are pushed to the object. I could be wrong since the constructor isn't being called here.

I'm just not sure if the `AssertionError` check around the `save()` will cover all our bases. Not certain, so I'd just like this confirmed. Thanks!

Reply:
I feel like I tested and found them invoked on save... I was mistaken, you are correct, it is checked on assignment.. changed.